torch/__init__.py

```diff
@@ -2065,6 +2065,28 @@ def _constrain_as_size(symbol, min: Optional[builtins.int] = None, max: Optional
     """
     torch.sym_constrain_range_for_size(symbol, min=min, max=max)
 
+def import_device_backends():
+    if sys.version_info < (3, 10):
+        from importlib_metadata import entry_points
+    else:
+        from importlib.metadata import entry_points
+
+    for backend in entry_points(group='torch.backends'):
+        try:
+            backend_hook = backend.load() 
+            if not hasattr(backend_hook, 'init_custom_backend'):
+                print(f"No explicit backend init function for \'{backend_hook.__name__}\' has been found, custom backend can't be loaded.")
+                continue
+            backend_hook.init_custom_backend()
+        except IndexError:
+            pass
```

Comment:
I think we don't need to call the init function, only load it.
Since the init function is implemented in your extension, your extension will be imported automatically after `backend.load()` is called. And calling it directly might be risky.

Here is an simple example and any suggestions are welcome.

in `torch_npu` implementation:

```python
# setup.py
setup(
    entry_points={
        'torch.backends': [
            'torch_npu = torch_npu:_init_device_backend',
        ],
    }
)

# torch_npu/__init__.py
def _init_device_backend():
    pass
```

in pytorch:

```python
# torch/__init__.py
for plugin in discovered_plugins:
    try:
        # just loads the plugin without calling
        plugin.load()
    except Exception:
        # keep quiet
        pass
```